src/yggdrasil/core.go

```diff
@@ -121,15 +111,13 @@ func (c *Core) addPeerLoop() {
 	}
 }
 
-// UpdateConfig updates the configuration in Core and then signals the
-// various module goroutines to reconfigure themselves if needed
+// UpdateConfig updates the configuration in Core with the provided
+// config.NodeConfig and then signals the various module goroutines to
+// reconfigure themselves if needed.
 func (c *Core) UpdateConfig(config *config.NodeConfig) {
 	c.log.Infoln("Reloading configuration...")
 
-	c.configMutex.Lock()
-	c.configOld = c.config
-	c.config = *config
-	c.configMutex.Unlock()
+	c.config.Replace(*config)
```

Comment:
Just a reminder that calling `Get` later is going to race, so it would be better to get the old config as a return arg, and then pass the old/new configs into the `reconfigure` channels, so other components know *exactly* what they need to compare.